plutus-core/plutus-ir/src/PlutusIR/Transform/RewriteRules/DecodeEncodeUtf8.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE GADTs           #-}
+{-# LANGUAGE LambdaCase      #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeOperators   #-}
+
+module PlutusIR.Transform.RewriteRules.DecodeEncodeUtf8
+  ( decodeEncodeUtf8
+  ) where
+
+
+import PlutusCore.Default
+import PlutusIR
+
+{- | This rewrites: `(decodeUtf8 (encodeUtf8 x)) => x`
+
+This rewrite is safe because x is either valid by construction *In Haskell, but not in GHC-CORE*
+(see <https://hackage.haskell.org/package/text-2.1/docs/Data-Text.html#g:2>)
```

Comment:
I'm not sure I understand the relevance of the link.

Reply:
I will address it in a new PR solely for this decode/encode rewriterule